ci_watson/artifactory_helpers.py

```diff
@@ -105,17 +117,22 @@ def get_bigdata(*args):
     /path/to/example.fits
 
     """
-    src = os.path.join(get_bigdata_root(), *args)
+    src = os.path.join(get_bigdata_root(repo=repo), *args)
```

Comment:
Your solution leads to inconsistent syntax for accessing data in the repository from one test to another.  The 'repo' parameter gets defined with a default in the base class (which can still be revised on the fly if needed) leaving the rest of the syntax identical for all tests.  Personally, 'explicit is better than implicit' and this was intended to be a case where this information was being made explicit.

Reply:
Another issue with implementing support for multiple repositories: this information ONLY applies to URLs, not to local directories.  Unfortunately, `get_bigdata()` does not distinguish and has no way to recognize the repo name for URLs when provided as part of the args.  This was demonstrated as I worked on an interim commit to remove use of this parameter from the `get_data()` method which calls `get_bigdata` in the JWST tests.   This led me to the current implementation of using a separate parameter for this information.

Reply:
> not to local directories

I am not sure I follow. I have no problem using the currently `ci-watson` (as released on PyPI before this PR) running both HSTCAL, `calcos`, and `acstools` tests locally without changing `TEST_BIGDATA`. Let's discuss when we meet. Perhaps we set things up differently at the local level.

Reply:
To enable someone to run tests for multiple repos that use Artifactory, my recommendation is to
```bash
export TEST_BIGDATA="https://bytesalad.stsci.edu/artifactory"
```

Reply:
Is anyone really going to be running tests for multiple repos from the same Jenkinsfile?

Right now we are setting this up for Jenkins runs for specific repos.  At the top of the Jenkinsfile we can specify `TEST_BIGDATA` to whatever we want.  We can do the same when running locally.

And in fact it would be best to be running the tests from a local cache that is updated from Artifactory instead of pulling all (~100GB for `jwst`) data from Artifactory every time the tests are run.  Though this is a separate issue that we should discuss.

Reply:
> Is anyone really going to be running tests for multiple repos from the same Jenkinsfile?

No. However, I do run big data tests for multiple repos on the same machine. In that case, I don't have to keep changing `TEST_BIGDATA` every time. IMHO it is the best that we keep local and Jenkins settings as similar as possible to reduce the element of surprise (and frustration).

Reply:
This feature was designed to support the tests for the cases when someone wants to write pipeline tests (and add them to the `tests_nightly` set of tests) which use the simulated data as input, data found in another repo on Artifactory.

Reply:
Didn't we discuss removing `repo` from the call as @pllim asked above?  I think this is cleaner.  It means that the tests just need to know which repo they pull from.  That's something that can be dealt with on the `jwst` side.

I guess I must have missed this is my summary comments of our discussion yesterday.

Reply:
I will take another look at the syntax involved to do this...

Reply:
This has now been fixed with the removal of 'repo' as an argument to `ci_watson.get_bigdata` in the latest `ci_watson` commit along with corresponding code changes here to the `base_classes` and `base_test` modules.

Reply:
It turns out that Artifactory will ONLY return a success when querying the URL if it points to a specific repo.  However, in order to support use of multiple repos within a single set of tests (through use of `input_repo` class attribute), we require that `TEST_BIGDATA` be defined only as "https://bytesalad.stsci.edu/artifactory".  Since this is the only value available to the `get_bigdata_root()` function, the query as to whether this is valid or not (`_is_url()` or `check_url()`) fails and everything breaks.  The **only way** to resolve this is to pass in the `input_repo` attrtibute value as an additional argument to `get_bigdata` and `get_bigdata_root`; namely, the `repo` parameter.

Therefore, the code has been restored to use the `repo` parameter AFTER making sure the tests ran successfully against BOTH Artifactory and against the local cache.

Reply:
That's odd.  Using the current master of `ci_watson`:
```
In [1]: from ci_watson import artifactory_helpers

In [2]: artifactory_helpers.get_bigdata_root()
Out[2]: 'https://bytesalad.stsci.edu/artifactory'

In [3]: artifactory_helpers.get_bigdata('jwst-pipeline', 'dev', 'nirspec/test_pipelines/truth/jwtest1013001_01101_00001_NRS1_x1d_ref.fits')
Out[3]: '/Users/jdavies/dev/jwst/jwtest1013001_01101_00001_NRS1_x1d_ref.fits'

In [4]: from jwst import datamodels

In [5]: file = artifactory_helpers.get_bigdata('jwst-pipeline', 'dev', 'nirspec/test_pipelines/truth/jwtest1013001_01101_00001_NRS1_x1d_ref.fits')

In [6]: with datamodels.open(file) as model:
   ...:     print(model.meta.filename)
   ...:     
jwtest1013001_01101_00001_NRS1_x1d_ref.fits
```

Reply:
I only made these changes after verifying that it would not work, and may still have a log file showing the errors (and debug statements as well).  Unfortunately, this is what I get under LInux for the master branch:

'''
(jwstrtconda) bash-4.2$ git checkout master
Switched to branch 'master'
(jwstrtconda) bash-4.2$ ipython
Python 3.6.6 |Anaconda, Inc.| (default, Jun 28 2018, 17:14:51) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from ci_watson import artifactory_helpers

In [2]: artifactory_helpers.get_bigdata_root?
Signature: artifactory_helpers.get_bigdata_root(envkey='TEST_BIGDATA')
Docstring: Find and returns the path to the nearest big datasets.
File:      /user/hack/dev/release/stsci_python/pypi/ci_watson/git/ci_watson/ci_watson/artifactory_helpers.py
Type:      function

In [3]: artifactory_helpers.get_bigdata_root()

In [4]: 

(jwstrtconda) bash-4.2$ export TEST_BIGDATA=https://bytesalad.stsci.edu/artifactory/jwst-pipeline/
(jwstrtconda) bash-4.2$ ipython
Python 3.6.6 |Anaconda, Inc.| (default, Jun 28 2018, 17:14:51) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from ci_watson import artifactory_helpers
   ...: 

In [2]: artifactory_helpers.get_bigdata_root()
   ...: 
Out[2]: 'https://bytesalad.stsci.edu/artifactory/jwst-pipeline/'

'''
The return value of None is what causes all the code to fail, and why I implemented the `repo` parameter.  I really wish it was different...

Reply:
Well that is odd!

I'm not running it from the repo, master branch. I'm just pip installing the release version
```
pip install ci-watson
```
Could it be that there's some odd behavior that is different when it is installed?

Reply:
Is your master branch out of date? If you look in the code, is it using requests.head or requests.get? Without seeing your git remotes and when you last synced them, I can't tell for sure. Will be happy to help on Monday.

Reply:
Also I'm not sure which version you're importing up there. You can try to print `ci_watson.__version__`. Anyway, I think we can get to the bottom of this on Monday when we are all in the same room staring at the same machine.